src/pages/Staking/index.tsx

```diff
@@ -1,10 +1,665 @@
+    isLoading: voteGovernanceTokenBalanceLoading,
+    data: voteGovernanceTokenBalance,
+    error: voteGovernanceTokenBalanceError,
+    refetch: voteGovernanceTokenBalanceRefetch
+  } = useQuery<MonetaryAmount<Currency<VoteUnit>, VoteUnit>, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'votingBalance',
+      address
+    ],
+    genericFetcher<MonetaryAmount<Currency<VoteUnit>, VoteUnit>>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(voteGovernanceTokenBalanceError);
+
+  // My Rewards
+  const {
+    isIdle: rewardAmountAndAPYIdle,
+    isLoading: rewardAmountAndAPYLoading,
+    data: rewardAmountAndAPY,
+    error: rewardAmountAndAPYError
+  } = useQuery<RewardAmountAndAPY, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getRewardEstimate',
+      address
+    ],
+    genericFetcher<RewardAmountAndAPY>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(rewardAmountAndAPYError);
+
+  // Estimated KINT Rewards & APY
+  const monetaryLockingAmount = newMonetaryAmount(lockingAmount, GOVERNANCE_TOKEN, true);
+  const {
+    isIdle: estimatedRewardAmountAndAPYIdle,
+    isLoading: estimatedRewardAmountAndAPYLoading,
+    data: estimatedRewardAmountAndAPY,
+    error: estimatedRewardAmountAndAPYError
+  } = useQuery<RewardAmountAndAPY, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getRewardEstimate',
+      address,
+      monetaryLockingAmount
+    ],
+    genericFetcher<RewardAmountAndAPY>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(estimatedRewardAmountAndAPYError);
+
+  const {
+    isIdle: stakedAmountAndEndBlockIdle,
+    isLoading: stakedAmountAndEndBlockLoading,
+    data: stakedAmountAndEndBlock,
+    error: stakedAmountAndEndBlockError
+  } = useQuery<StakedAmountAndEndBlock, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getStakedBalance',
+      address
+    ],
+    genericFetcher<StakedAmountAndEndBlock>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(stakedAmountAndEndBlockError);
+
+  const initialStakeMutation = useMutation<void, Error, LockingAmountAndUnlockHeight>(
+    (variables: LockingAmountAndUnlockHeight) => {
+      return window.bridge.interBtcApi.escrow.createLock(variables.amount, variables.unlockHeight);
+    },
+    {
+      onSuccess: (_, variables) => {
+        voteGovernanceTokenBalanceRefetch();
+        console.log('[initialStakeMutation onSuccess] variables => ', variables);
+        reset({
+          [LOCKING_AMOUNT]: '0.0',
+          [LOCK_TIME]: '0'
+        });
+      },
+      onError: error => {
+        // TODO: should add error handling UX
+        console.log('[initialStakeMutation onError] error => ', error);
+      }
+    }
+  );
+
+  const moreStakeMutation = useMutation<void, Error, LockingAmountAndUnlockHeight>(
```

Comment:
The issue you are seeing in testing is coming from batching the transactions even if you only want to update one of the two parts (increasing the locked amount OR extending the lock time).

> When trying to extend the lock time more without increasing the locked amount, it fails with this console log.
![image](https://user-images.githubusercontent.com/14966470/156105786-92ede91d-1dbd-4ff6-9728-f1f24e66c0b3.png)

If the user only wants to extend the lock-time, you will have to only call `window.bridge.interBtcApi.api.tx.escrow.increaseUnlockHeight(variables.unlockHeight)` or the interbtc-js higher-level `increaseUnlockHeight` https://github.com/interlay/interbtc-api/blob/master/src/parachain/escrow.ts#L132.

> When trying to increase the locked amount more without extending the lock time, it fails with this console log.
![image](https://user-images.githubusercontent.com/14966470/156105964-b16b608a-4fd8-46ea-a881-06de35fcbce3.png)

Similarly, when only increasing the locked amount, you have to call only `window.bridge.interBtcApi.api.tx.escrow.increaseAmount(
            variables.amount.toString(variables.amount.currency.rawBase)
          ),` or `increaseAmount` in https://github.com/interlay/interbtc-api/blob/master/src/parachain/escrow.ts#L127

Reply:
@nud3l 
I implemented them but it looks like there are problems with:
2. Only extend lock time (failing)
3. Increase amount and extend lock time (keeps loading)
Could you please have a look at them? UI code-wise